app/src/main/java/com/soldevcode/composechat/models/platform/audio/RecordingManager.kt

```diff
@@ -0,0 +1,189 @@
+package com.soldevcode.composechat.models.platform.audio
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.google.api.gax.rpc.ClientStream
+import com.google.api.gax.rpc.ResponseObserver
+import com.google.api.gax.rpc.StreamController
+import com.google.cloud.speech.v1.RecognitionConfig
+import com.google.cloud.speech.v1.SpeechClient
+import com.google.cloud.speech.v1.SpeechRecognitionAlternative
+import com.google.cloud.speech.v1.StreamingRecognitionConfig
+import com.google.cloud.speech.v1.StreamingRecognitionResult
+import com.google.cloud.speech.v1.StreamingRecognizeRequest
+import com.google.cloud.speech.v1.StreamingRecognizeResponse
+import com.google.protobuf.ByteString
+import com.soldevcode.composechat.presentation.MainViewModel
+import com.soldevcode.composechat.util.SpeechCredentialsProvider
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/** Referring to Google documentation
+https://cloud.google.com/speech-to-text
+/docs/transcribe-streaming-audio#perform_streaming_speech_recognition_on_an_audio_stream
+ */
+
+@SuppressLint("MissingPermission")
+@Composable
+fun rememberRecordingManager(viewModel: MainViewModel = viewModel()): RecordingManager {
+    val context = LocalContext.current
+    val scope = rememberCoroutineScope()
+    val showRecording = rememberSaveable { mutableStateOf(false) }
+    val responseObserver: ResponseObserver<StreamingRecognizeResponse?>? = null
+    val transcriptions = mutableListOf<String>()
+    val isRecording = rememberSaveable { mutableStateOf(false) }
+    val request: StreamingRecognizeRequest? = null
+    val audioRecord = AudioRecord(
+        MediaRecorder.AudioSource.MIC,
+        44100,
+        AudioFormat.CHANNEL_IN_MONO,
+        AudioFormat.ENCODING_PCM_16BIT,
+        AudioRecord.getMinBufferSize(
+            44100, AudioFormat.CHANNEL_IN_MONO, AudioFormat.ENCODING_PCM_16BIT
+        )
+    )
+
+    return remember {
+        RecordingManager(
+            context, showRecording, scope,
+            responseObserver, transcriptions,
+            isRecording, request, audioRecord, viewModel
```

Comment:
You shouldn't pass a viewModel through the manager :) Just move the textToSpeech into Compose or just use a callback that sets that value into the viewModel through Compose 👍🏻

Reply:
I see it now... passing the viewmodel to ui twice ...hoops.  I think I will try both to practice. Thanks for the review